lib/mayaUsd/ufe/Global.cpp

```diff
@@ -95,9 +96,17 @@ void exitingCallback(void* /* unusedData */)
     MayaUsd::ufe::finalize(/* exiting = */ true);
 }
 
+//------------------------------------------------------------------------------
+// Global variables
+//------------------------------------------------------------------------------
+
 int gRegistrationCount = 0;
 
 MCallbackId gExitingCbId = 0;
+
+// Subject singleton for observation of all USD stages.
+MayaUsd::ufe::MayaStagesSubject::RefPtr g_StagesSubject;
```

Comment:
Julien found a problem with the lifetime of the StagesSubject and the destructor of derived not being called.  If you look at the commits separately the first commit is all out the lifetime.